### PR TITLE
LibVCX fixes in Java wrapper

### PR DESCRIPTION
**Pull request by airoasis**
*Bug fix*
* The API connectionGetTheirPwDid returns pw_did not their_pw_did

*New APIs*
 * DisclosedProofApi.proofUpdateStateWithMessage
 * VcxApi.vcxUpdateInstitutionInfo

*Remove obsolete APIs*
* vcx_reset
* DisclosedProofApi.proofCreate

### CODE REVIEW

File: wrappers/java/src/main/java/com/evernym/sdk/vcx/proof/DisclosedProofApi.java [hunk 1]
```diff
@@ -17,35 +17,6 @@ private DisclosedProofApi() {
...
-        @SuppressWarnings({"unused", "unchecked"})
-        public void callback(int commandHandle, int err, int proofHandle) {
-            logger.debug("callback() called with: commandHandle = [" + commandHandle + "], err = [" + err + "], proofHandle = [" + proofHandle + "]");
-            CompletableFuture<Integer> future = (CompletableFuture<Integer>) removeFuture(commandHandle);
-            if (!checkCallback(future, err)) return;
-            future.complete(proofHandle);
-        }
-    };
-
-    public static CompletableFuture<CreateProofMsgIdResult> proofCreate(
```
Comment (Patrik-Stas) [inline wrappers/java/src/main/java/com/evernym/sdk/vcx/proof/DisclosedProofApi.java:30]: Is there a problem with this API? Why is this deleted? I believe we are actually using this in our app.

Comment (airoasis) [reply wrappers/java/src/main/java/com/evernym/sdk/vcx/proof/DisclosedProofApi.java:30]: The one we use for creating a disclosed proof is https://github.com/AbsaOSS/libvcx/blob/f64f763c005465970e6bc05949b756db0bebaf48/wrappers/java/src/main/java/com/evernym/sdk/vcx/proof/DisclosedProofApi.java#L320. The method signature of `vcx_disclosed_proof_create_with_request` is https://github.com/AbsaOSS/libvcx/blob/f64f763c005465970e6bc05949b756db0bebaf48/libvcx/src/api/disclosed_proof.rs#L89. and https://github.com/AbsaOSS/libvcx/blob/f64f763c005465970e6bc05949b756db0bebaf48/wrappers/java/src/main/java/com/evernym/sdk/vcx/proof/DisclosedProofApi.java#L44 doesn't correspond with this. Only the Java wrapper has this obsolete method definition.

**Comment by mirgee** [comment]
Delaying merge until we are able to import resulting library without segfaults.
